File: CSC205-Cpp/Module06-Arrays-ArrayLists-Collections-and-Iterators/debug01/remove_evens.cpp
// remove_evens.cpp
// Removes every even number from a vector, leaving the odd numbers in
// their original order.

#include <iostream>
#include <vector>
using namespace std;

void printVector(const vector<int>& v) {
    cout << "[";
    for (size_t i = 0; i < v.size(); i++) {
        if (i > 0) {
            cout << ", ";
        }
        cout << v[i];
    }
    cout << "]" << endl;
}

// Erases each even element from v.
void removeEvens(vector<int>& v) {
    for (size_t i = 0; i < v.size(); i++) {
        if (v[i] % 2 == 0) {
            v.erase(v.begin() + i);
        }
    }
}

int main() {
    vector<int> numbers = { 2, 4, 6, 7, 8, 10, 11, 12 };

    cout << "Before: ";
    printVector(numbers);

    removeEvens(numbers);

    cout << "After:  ";
    printVector(numbers);
    cout << "Remaining count: " << numbers.size() << endl;
    return 0;
}

File: CSC205-Java/_solutions/Module06-Arrays-ArrayLists-Collections-and-Iterators/debug01/RemoveEvens.java
import java.util.ArrayList;

// Removes every even number from a list of integers.
public class RemoveEvens {
    public static void removeEvens(ArrayList<Integer> values) {
        for (int i = 0; i < values.size(); i++) {
            if (values.get(i) % 2 == 0) {
                values.remove(i);
                i--;
            }
        }
    }

    public static void main(String[] args) {
        ArrayList<Integer> values = new ArrayList<>();
        int[] data = {3, 8, 6, 10, 7, 12, 14, 5, 2};
        for (int d : data) {
            values.add(d);
        }

        System.out.println("Before: " + values);
        removeEvens(values);
        System.out.println("After:  " + values);
    }
}
